maven-build-caching-samples/animal-sniffer-project/pom.xml

```diff
@@ -0,0 +1,125 @@
+        <executions>
+          <execution>
+            <id>animal-sniffer-check</id>
+            <phase>test</phase>
+            <goals>
+              <goal>check</goal>
+            </goals>
+          </execution>
+        </executions>
+
+        <configuration>
+          <signature>
+            <groupId>org.codehaus.mojo.signature</groupId>
+            <artifactId>java15</artifactId>
+            <version>1.0</version>
+          </signature>
+        </configuration>
+      </plugin>
+    </plugins>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>com.gradle</groupId>
+          <artifactId>gradle-enterprise-maven-extension</artifactId>
+          <configuration>
+            <gradleEnterprise>
+              <plugins>
+                <plugin>
+                  <groupId>org.codehaus.mojo</groupId>
+                  <artifactId>animal-sniffer-maven-plugin</artifactId>
+                  <executions>
+                    <execution>
+                      <id>animal-sniffer-check</id>
+                      <inputs>
+                        <ignoredProperties>
+                          <ignore>localRepository</ignore>
+                          <ignore>project</ignore>
+                        </ignoredProperties>
+                        <fileSets>
+                          <fileSet>
+                            <name>outputDirectory</name>
+                            <normalization>CLASSPATH</normalization>
+                          </fileSet>
+                          <fileSet>
+                            <name>testOutputDirectory</name>
+                            <normalization>CLASSPATH</normalization>
+                          </fileSet>
+                        </fileSets>
+                        <properties>
+                          <property>
+                            <name>ignores</name>
+                          </property>
+                          <property>
+                            <name>annotations</name>
+                          </property>
+                          <property>
+                            <name>ignoreDependencies</name>
+                          </property>
+                          <property>
+                            <name>includeDependencies</name>
+                          </property>
+                          <property>
+                            <name>excludeDependencies</name>
+                          </property>
+                          <property>
+                            <name>checkTestClasses</name>
+                          </property>
+                          <property>
+                            <name>skip</name>
+                          </property>
+                          <property>
+                            <name>failOnError</name>
+                          </property>
+                        </properties>
+                      </inputs>
+                      <nestedProperties>
```

Comment:
i like the nestedproperties example! i was always confused how exactly these work.